pkg/driver/controller_server.go

```diff
@@ -457,6 +455,38 @@ func (driver *Driver) createVolume(
 				return nil, status.Error(codes.Internal, "Parent volume ID is missing in the snapshot response")
 			}
 
+			// Get the parent volume from the snapshot
+			existingParentVolume, err := storageProvider.GetVolume(existingSnap.VolumeID)
+			if err != nil {
+				log.Error("err: ", err.Error())
+				return nil,
+					status.Error(codes.Internal,
+						fmt.Sprintf("Failed to check if snapshot's parent volume %s with ID %s exist, err: %s",
+							existingSnap.VolumeName, existingSnap.VolumeID, err.Error()))
+			}
+			// The requested size is must be at least equal to the snapshot's parent volume size
+			if size < existingParentVolume.Size {
```

Comment:
This condition is checked in the CSP so can be removed i.e the size check

Reply:
Got it !